src/tabletraits.jl

```diff
@@ -0,0 +1,72 @@
+using TableTraits
+
+TableTraits.isiterable(x::IndexedTable) = true
+TableTraits.isiterabletable(x::IndexedTable) = true
+
+function TableTraits.getiterator{S<:IndexedTable}(source::S)
+    return rows(source)
+end
+
+# Sink
+
+@generated function _fillIndexedTable{idx_indices,data_indices}(iter,idx_storage,data_storage,::Type{idx_indices},::Type{data_indices})
+    push_exprs = Expr(:block)
+    for (i,idx) in enumerate(map(i->i.parameters[1],idx_indices.parameters))
+        ex = :( push!(idx_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    for (i,idx) in enumerate(map(i->i.parameters[1],data_indices.parameters))
+        ex = :( push!(data_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    quote
+        for row in iter
+            $push_exprs
+        end
+    end
+end
+
+function IndexedTable(x; idxcols::Union{Void,Vector{Symbol}}=nothing, datacols::Union{Void,Vector{Symbol}}=nothing)
```

Comment:
Yes, you should just be able to add another method that handles that case, right? It would be good if the named arguments had the same semantics, of course.

I'm also not sure this is the right API, I just was loosely inspired by [this](https://github.com/JuliaComputing/NDSparseIO.jl/blob/master/src/NDSparseIO.jl#L29).